sklearn/utils/validation.py

```diff
@@ -212,6 +213,26 @@ def check_consistent_length(*arrays):
                          " samples: %r" % [int(l) for l in lengths])
 
 
+def _convert_iterable(iterable):
+    """Helper convert iterable to arrays of sparse matrices.
```

Comment:
This is wrong. Here is an alternative suggestion:

```suggestion
    """Ensure iterable supports fancy indexing or convert to an indexable variant.
```

Reply:
Actually fancy indexing is not even guaranteed at this point because lists pass unchanged and do not support fancy indexing. So this is just "indexable" or "support indexing".